lib/active_monitoring/runtime/flow.ex

```diff
@@ -70,19 +85,14 @@ defmodule ActiveMonitoring.Runtime.Flow do
     Call |> Repo.get_by(sid: call_sid)
   end
 
-  defp fetch_subject(from, campaign_id) do
-    Subject |> Repo.get_by(phone_number: from, campaign_id: campaign_id)
+  defp fetch_subject(registration_identifier, campaign_id) do
+    Subject |> Repo.get_by(registration_identifier: registration_identifier, campaign_id: campaign_id)
   end
 
   defp fetch_or_insert_call(call_sid, from, campaign_id) do
     case fetch_call(call_sid) do
       nil ->
-        case fetch_subject(from, campaign_id) do
-          nil ->
-            Call.changeset(%Call{}, %{sid: call_sid, campaign_id: campaign_id, from: from}) |> Repo.insert!
-          subject ->
-            Call.changeset(%Call{}, %{sid: call_sid, subject_id: subject.id, campaign_id: campaign_id, from: from}) |> Repo.insert!
-        end
+        Call.changeset(%Call{}, %{sid: call_sid, campaign_id: campaign_id, from: from}) |> Repo.insert!
```

Comment:
It's usually a good idea to encapsulate changeset builders in a function that hides structure and has a more semantically significative name (like `new_call`)